src/rpcrawtransaction.cpp

```diff
@@ -24,111 +23,280 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
-    res["Interest"]=bb.InterestSubsidy;
-    res["BoincReward"]=bb.ResearchSubsidy;
-    res["Difficulty"]=GetBlockDifficulty(block.nBits);
+    if (bb.Magnitude > 0)
+    {
+        res.push_back(std::make_pair("Magnitude", RoundToString(bb.Magnitude,8)));
+        res.push_back(std::make_pair("Boinc Reward", RoundToString(bb.ResearchSubsidy,8)));
+    }
 
+    res.push_back(std::make_pair("Is Superblock", (bb.superblock.length() >= 20 ? "true" : "false")));
 
     if(fDebug)
     {
-        res["xbbBoincPublicKey"]=bb.BoincPublicKey;
+        if (bb.superblock.length() >= 20)
+            res.push_back(std::make_pair("Neural Contract Binary Size", ToString(bb.superblock.length())));
+
+        res.push_back(std::make_pair("Organization", bb.Organization));
+        res.push_back(std::make_pair("Client Version", bb.clientversion));
+        res.push_back(std::make_pair("Neural Hash", bb.NeuralHash));
+        res.push_back(std::make_pair("CurrentNeuralHash", bb.CurrentNeuralHash));
+        res.push_back(std::make_pair("Boinc Public Key", bb.BoincPublicKey));
+    }
 
-        res["xbbOrganization"]=bb.Organization;
-        res["xbbClientVersion"]=bb.clientversion;
+    return res;
+}
 
-        res["xbbNeuralHash"]=bb.NeuralHash;
-        res["xbbCurrentNeuralHash"]=bb.CurrentNeuralHash;
-        res["xbbNeuralContractSize"]=bb.superblock.length();
-    }
-    else
+std::vector<std::pair<std::string, std::string>> GetTxNormalBoincHashInfo(const CMerkleTx& mtx)
+{
+    assert(!mtx.IsCoinStake() && !mtx.IsCoinBase());
+    std::vector<std::pair<std::string, std::string>> res;
+
+    try
     {
-        if(bb.superblock.length()>=20)
+        const std::string &msg = mtx.hashBoinc;
+
+        res.push_back(std::make_pair("Network Time", TimestampToHRDate((double)mtx.nTime)));
+
+        if (fDebug)
+            res.push_back(std::make_pair("Message Length", ToString(msg.length())));
+
+        std::string sMessageType = ExtractXML(msg, "<MT>", "</MT>");
+        std::string sTxMessage = ExtractXML(msg, "<MESSAGE>", "</MESSAGE>");
+        std::string sRainMessage = ExtractXML(msg, "<NARR>", "</NARR>");
+
+        if (sMessageType.length())
+        {
+            res.push_back(std::make_pair("Message Type", "Contract"));
+
+            if (sMessageType == "beacon")
+            {
+                res.push_back(std::make_pair("Contract Type", "Beacon"));
+
+                std::string sBeaconCommit = ExtractXML(msg, "<MA>", "</MA>");
+
+                if (sBeaconCommit == "A")
+                {
+                    std::string sBeaconCPID = ExtractXML(msg, "<MK>", "</MK>");
+                    std::string sBeaconEncodedContract = ExtractXML(msg, "<MV>", "</MV>");
+
+                    if (sBeaconEncodedContract.length() < 256)
+                    {
+                        // If for whatever reason the contract is not a proper one and the average length does exceed this size; Without this a seg fault will occur on the DecodeBase64
```

Comment:
This is pretty serious. Do you think such segfault could occur in main.cpp while processing network transaction? Please post as separate issue!

Reply:
it is serious if the beacon contract mv is 1 and not a encoded contract